qemu/tests/qemu_guest_agent.py

```diff
@@ -818,6 +818,31 @@ def _action_after_fsfreeze(self, *args):
             raise error.TestFail("FS freeze failed, guest still can"
                                  " write file")
 
+    def gagent_check_reset_reboot_shutdown(self, test, params, env):
+
+        """
+        Execute "system_reset" and send "shutdown,reboot"
+        command to guest agent after FS freezed
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        vm = env.get_vm(params["main_vm"])
+        vm.verify_alive()
+        gagent = self.gagent
+        gagent.fsfreeze()
+
+        if gagent.shutdown(gagent.SHUTDOWN_MODE_POWERDOWN):
+            test.fail("shutdown command shouldn't succeed for freeze FS")
```

Comment:
when guest is frozen, shutdown command will return msg like "The command xxx has been disabled for this instance"

shutdown may also fail due to other unexpected reasons.

It's better to check whether it's due to fs is frozen or not.

Reply:
@lijinlijin  You are right.
1.  This message "The command xxx has been disabled for this instance"  is from VAgentCmdError, so except VAgentCmdError in shutdown function.  Anyway, shutdown command doesn't concern Error message.  But must make sure shutdown command cann't success. 
2. In order to make sure it's failure is from frozen.  I will execute verify_fsfreeze_status(gagent.FSFREEZE_STATUS_FROZEN) before powerdown, reboot and halt.   function fsfreeze includes verify_fsfreeze_status()

Reply:
Thanks Lijin.

Reply:
Append,   fsfreeze() function includes verify_fsfreeze_status(gagent.FSFREEZE_STATUS_FROZEN)